staging/src/k8s.io/apiserver/pkg/storage/etcd3/featuresupportchecker/feature_support_checker.go

```diff
@@ -0,0 +1,166 @@
+/*
+Copyright 2024 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package featuresupportchecker
```

Comment:
No need to create a separate package. From import perspective it's more readable to have `etcd3.Supports(etcd.RequestWatchProgress)` than `featuresupportchecker.Supports(etcd.RequestWatchProgress)`.

Reply:
This was because of import cycle inside flowcontrol.

Reply:
ok, maybe name `etcdfeature` ? The import would be `etcdfeature.Supports(storage.RequestWatchProgress)`?